src/core/Rational.pm6

```diff
@@ -236,16 +245,17 @@ my role Rational[::NuT = Int, ::DeT = ::("NuT")] does Real {
 
         # Round the final number, based on the remaining fractional part
         if 2*$fract >= 1 {
-            for @fract-digits-1 ... 0 -> $n {
+            for @fract-digits - 1 ... 0 -> $n {
```

Comment:
I don't know how big `@fract-digits` gets, but `.reverse` (e.g., `(^@fract-digits).reverse`) is much faster than a decreasing `...`.

Reply:
In general, yes. In this specific case, it is exceedingly rare that it iterates more than 3-4 times before short circuiting and exiting so the iterator only needs to produce 3-4 elements. '(^@fract-digits).reverse)' will require that all the elements be generated every time. In testing, '@fract-digits - 1 ... 0' is slightly faster most of the time.